Adafruit_MQTT.cpp

```diff
@@ -213,13 +213,15 @@ uint16_t Adafruit_MQTT::processPacketsUntil(uint8_t *buffer, uint8_t waitforpack
       break;
     }
 
-    if ((buffer[0] >> 4) == waitforpackettype)
-    {
-      return len;
-    }
-    else
-    {
-      ERROR_PRINTLN(F("Dropped a packet"));
+    uint8_t packetType = (buffer[0] >> 4);
+    if (packetType == waitforpackettype) {
+        return len;
+    } else {
+        if (packetType == MQTT_CTRL_PUBLISH) {
```

Comment:
I've tested this and it seems to work. May want to handle other packet types this way in the future by adding callbacks for them rather than handling it all synchronously?

Reply:
Could you explain how you tested this? Thanks!

Reply:
Sure. I had an MQTT broker running on my machine as well as an MQTT client that I could trigger to start sending requests to the server as fast as it'd handle it.

I then had an ESP32 subscribing to this topic where the loop code looked something like this:
```void loop() {
    ArduinoOTA.handle();
    connectMqtt();
    mqtt.processPackets(100);

   // Do some other processing
   delay(500);

    if (!mqtt.ping(20)) {
        Serial.println("Ping failed!");
        mqtt.disconnect();
    }
}```

Without this change it would drop all the incoming `PUBLISH` packets. With this change it handles the `PING` and `PUBLISH` packets intermixed just fine.

Reply:
What's ArduinoOTA?

Reply:
Oh, I was using that to program it over WiFi without having to plug the board into USB. It has no bearing on the testing - will remove it from the sample code.